ospfd/ospf_gr_helper.c

```diff
@@ -0,0 +1,982 @@
+/*
+ * OSPF Gracefull Retsart helper functions.
+ *
+ * Copyright (C) 2020-21 Vmware, Inc.
+ * Rajesh Kumar Girada
+ *
+ * This file is part of GNU Zebra.
+ *
+ * GNU Zebra is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * GNU Zebra is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include "thread.h"
+#include "memory.h"
+#include "linklist.h"
+#include "prefix.h"
+#include "if.h"
+#include "table.h"
+#include "vty.h"
+#include "filter.h"
+#include "log.h"
+#include "jhash.h"
+
+#include "ospfd/ospfd.h"
+#include "ospfd/ospf_interface.h"
+#include "ospfd/ospf_asbr.h"
+#include "ospfd/ospf_lsa.h"
+#include "ospfd/ospf_lsdb.h"
+#include "ospfd/ospf_neighbor.h"
+#include "ospfd/ospf_spf.h"
+#include "ospfd/ospf_flood.h"
+#include "ospfd/ospf_route.h"
+#include "ospfd/ospf_zebra.h"
+#include "ospfd/ospf_dump.h"
+#include "ospfd/ospf_errors.h"
+#include "ospfd/ospf_nsm.h"
+#include "ospfd/ospf_ism.h"
+#include "ospfd/ospf_gr_helper.h"
+
+const char *ospf_exit_reason_desc[] = {
+	"Unknown reason",
+	"Helper inprogress",
+	"Topology Change",
+	"Grace timer expairy",
+	"Successful graceful restart",
+};
+
+const char *ospf_restart_reason_desc[] = {
+	"Unknown restart",
+	"Software restart",
+	"Software reload/upgrade",
+	"Switch to redundant control processor",
+};
+
+const char *ospf_rejected_reason_desc[] = {
+	"Unknown reason",
+	"Helper support disabled",
+	"Neighbour is not in FULL state",
+	"Supports only planned restart but received for unplanned",
+	"Topo change due to change in lsa rxmt list",
+	"LSA age is more than Grace interval",
+};
+
+static void ospf_grace_lsa_show_info(struct vty *vty, struct ospf_lsa *lsa,
+				     json_object *json);
+static bool ospf_check_chg_in_rxmt_list(struct ospf_neighbor *nbr);
```

Comment:
The abbreviation chg for 'change' is very unusual for me. I can see this in the PR very often. Is that a thing in FRR I didn't know about?

Reply:
Used abbreviation to make the function name shorter. Anyhow, as suggested changed it to 'change'.